edi_sale_utm_oca/tests/test_sale_order.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+
+from odoo.tests import tagged
+
+from odoo.addons.base.tests.common import BaseCommon
+from odoo.addons.edi_oca.tests.common import EDIBackendTestMixin
+from odoo.addons.edi_sale_oca.tests.common import OrderMixin
+
+
+@tagged("post_install", "-at_install")
+class TestSaleOrder(BaseCommon, OrderMixin, EDIBackendTestMixin):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls._setup_records()
+
+    def test_source_edi(self):
+        exchange_record = self.env["edi.exchange.record"].create(
+            {
+                "backend_id": self.backend.id,
+                "type_id": self.exchange_type_in.id,
+                "model": "an.other.model.with.same.id",
+            }
+        )
+
+        vals = {
+            "name": "",
+            "origin_exchange_record_id": exchange_record.id,
+            "partner_id": self.partner.id,
+        }
+
+        sale_order = self._create_sale_order(**vals)
+
+        self.assertEqual(
+            sale_order.edi_source_id.id,
+            self.env.ref("edi_utm_oca.utm_source_edi").id,
+        )
```

Comment:
You can test the same with [this test](https://github.com/OCA/edi-framework/blob/16.0/edi_sale_oca/tests/test_order.py)